### PR TITLE
Add AUTHENTICATION_TYPE column to IDP_AUTHENTICATOR table.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Issue:
- https://github.com/wso2/product-is/issues/21473

With this PR, the following changes are introduced:
- A new enum, AUTHENTICATION_TYPE, with values IDENTIFICATION and VERIFICATION.
1. IDENTIFICATION: Collect the user identifier and authenticate
2. VERIFICATION: Only perform extra factor verification of a given authenticated user.
- Addition of a new column, AUTHENTICATION_TYPE, to the IDP_AUTHENTICATOR table.
- Updated insert queries. Note all the existing code that saves authenticators to the above table falls under IDENTIFICATION.

Note: The property value is VERIFICATION, additional code logic is required to add the 2FA tag to the configuration tag list. As all existing code logics save IDENTIFICATION type authenticators not need to update the read queries, as there is no logic to be exected if type is IDENTIFICATION.

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 1]
```diff
@@ -1427,6 +1428,7 @@ public void addFederatedAuthenticatorConfig(FederatedAuthenticatorConfig authnCo
             prepStmt1.setString(4, authnConfig.getName());
             prepStmt1.setString(5, authnConfig.getDisplayName());
             prepStmt1.setString(6, authnConfig.getDefinedByType().toString());
+            prepStmt1.setString(7, AuthenticationType.IDENTIFICATION.toString());
```
Comment (malithie) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:1431]: better add a comment mentioning federated authenticators are always supposed authenticate an externally managed user and share user identity and attributes. Therefore, by default these are always of 'identification' type

File: components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java [hunk 2]
```diff
@@ -28,4 +28,15 @@ public static enum DefinedByType {
         SYSTEM,
         USER
     }
+
+    /**
+     * The Authentication Type -
+     * Identification: Can collect the user identifier and authenticator. Can engage in any step of the login flow.
```
Comment (malithie) [inline components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java:34]: Can collect the user identifier, credential and authenticate.

File: components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java [hunk 3]
```diff
@@ -28,4 +28,15 @@ public static enum DefinedByType {
...
+     * Verification: Only perform additional verification. Can engage in 2nd or later step of the login flow.
```
Comment (malithie) [inline components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java:35]: Only perfom addition verification on top of a user identity provided as input

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 4]
```diff
@@ -5139,6 +5140,7 @@ private int addAuthenticator(Connection conn, int tenantId, String idpName, Stri
             prepStmt.setString(5, "1");
             prepStmt.setString(6, authenticatorDispalyName);
             prepStmt.setString(7, definedByType);
+            prepStmt.setString(8, AuthenticationType.IDENTIFICATION.toString());
```
Comment (malithie) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:5143]: Can we say by default unless specified authentication type is identified as of 'IDENTIFICATION' ?

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:1433]: Addressed with commit c4009b8301f34d84dcfd97dfe775da67c9fd14d8

File: components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java]: Addressed with commit c4009b8301f34d84dcfd97dfe775da67c9fd14d8

File: components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java (hunk 3 above)
Comment (Thisara-Welmilla) [reply components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/AuthenticatorPropertyConstants.java]: Addressed with commit c4009b8301f34d84dcfd97dfe775da67c9fd14d8

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 4 above)
Comment (Thisara-Welmilla) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:5144]: Addressed with commit c4009b8301f34d84dcfd97dfe775da67c9fd14d8

**Review by malithie — APPROVED** [review]

**Pull request merged**